common/src/api/external/http_pagination.rs

```diff
@@ -871,4 +931,61 @@ mod test {
         assert_eq!(data_page.direction, PaginationOrder::Ascending);
         assert_eq!(data_page.limit, limit);
     }
+
+    #[test]
+    fn test_scan_by_time_and_id() {
+        let scan = ScanByTimeAndId {
+            sort_by: TimeAndIdSortMode::Ascending,
+            selector: (),
+        };
+
+        let list = list_of_things();
+        let item0_time = list[0].identity.time_created;
+        let item0_id = list[0].identity.id;
+        let item0_marker = (item0_time, item0_id);
+
+        let last_idx = list.len() - 1;
+        let item_last_time = list[last_idx].identity.time_created;
+        let item_last_id = list[last_idx].identity.id;
+        let item_last_marker = (item_last_time, item_last_id);
+
+        let marker_fn =
+            |_: &ScanByTimeAndId, item: &MyThing| -> (DateTime<Utc>, Uuid) {
+                (item.identity.time_created, item.identity.id)
+            };
+        let (p0, p1) = test_scan_param_common(
+            &list,
+            &scan,
+            "sort_by=ascending",
+            &item0_marker,
+            &item_last_marker,
+            &scan,
+            &marker_fn,
+        );
+
+        assert_eq!(scan.direction(), PaginationOrder::Ascending);
+
+        // Verify data pages based on the query params.
+        let limit = NonZeroU32::new(123).unwrap();
+        let data_page = data_page_params_with_limit(limit, &p0).unwrap();
+        assert_eq!(data_page.marker, None);
+        assert_eq!(data_page.direction, PaginationOrder::Ascending);
+        assert_eq!(data_page.limit, limit);
+
+        let data_page = data_page_params_with_limit(limit, &p1).unwrap();
+        assert_eq!(data_page.marker, Some(&item_last_marker));
+        assert_eq!(data_page.direction, PaginationOrder::Ascending);
+        assert_eq!(data_page.limit, limit);
+
+        // Test error case
+        let error = serde_urlencoded::from_str::<PaginatedByTimeAndId>(
+            "sort_by=descending",
+        )
+        .unwrap_err();
+
+        assert_eq!(
+            error.to_string(),
+            "unknown variant `descending`, expected `ascending`"
+        );
+    }
```

Comment:
These tests are all kinda meh to me but they run instantly and it was easy to follow the pattern of the others.

Reply:
I take it back, they caught a mistake I made.